src/reconcile/rec_visibility.c

```diff
@@ -436,6 +418,25 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         }
     }
 
+    /*
+     * If we found a tombstone with a time pair earlier than the update it applies to, which can
+     * happen if the application performs operations with timestamps out-of-order, make it invisible
+     * by making the start time pair match the stop time pair of the tombstone. We don't guarantee
+     * that older readers will be able to continue reading content that has been made invisible by
+     * out-of-order updates.
+     */
+    if (upd_select->stop_ts < upd_select->start_ts ||
+      (upd_select->stop_ts == upd_select->start_ts &&
+          upd_select->stop_txn < upd_select->start_txn)) {
```

Comment:
`upd_select->stop_txn <= upd_select->start_txn` ?  Why in hs.c, it is `stop_ts_pair.txnid <= upd->txnid`?

Reply:
The difference is that in `hs.c`, we're deciding whether an update needs to be inserted into history.  There if `stop <= start`, the update can never be read so it doesn't need to be stored.

In `rec_visibility.c`, we're saying `if stop < start, set start = stop`.  I can make that `if stop <= start, set start = stop` but then we'd generate a warning for the (odd, but allowed) situation where a transaction does an insert and then a remove of the same record.

Reply:
Thanks, that make sense.

Reply:
May be worth to add a comment here to indicate it is allowed that a transaction does an insert and then a remove of the same record.